runtime/src/bank.rs

```diff
@@ -2405,7 +2465,16 @@ impl Bank {
                 account.lamports += rent_to_be_paid;
                 self.store_account(&pubkey, &account);
             });
-        leftover_lamports
+
+        if enforce_fix {
+            assert_eq!(leftover_lamports, 0);
+        } else if leftover_lamports != 0 {
+            warn!(
```

Comment:
Not a blocker, but something to think about for the future.  Perhaps we should be storing capitalization change events like this with more persistence?

Reply:
@t-nelson Well, once `enfoced_fix` is true, non-zero `leftover_lamports` is a very hard error: 
https://github.com/solana-labs/solana/pull/12112/files#diff-a7549f152920d85fb44e6a784b8e5e1fR2470
Also, capitalization is now periodically checked when creating a new snapshot #11927 

You have a point for alerting by events, but we'll have the most hardest one (dead of cluster) if ever this rent distribution has more leftover or bad capitalization in general.

This lethal nature is justified because this is a possible indication of attempted unauthorized transfer of tokens. (ie security breach, or frankly a theft)

Reply:
Yeah sure!  We'll probably notice the cluster halting soon enough :wink: